### PR TITLE
⚠️ CONFLICT! Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request: CONFLICT #

Achtung!!!

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-generic.git`](https://github.com/cisagov/skeleton-generic.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Take ownership of this pull request by removing any other assignees.

1. Clone the repository locally, and reapply the merge:

    ```console
    git clone git@github.com:cisagov/skeleton-packer.git skeleton-packer
    cd skeleton-packer
    git remote add skeleton https://github.com/cisagov/skeleton-generic.git
    git remote set-url --push skeleton no_push
    git switch develop
    git switch --create lineage/skeleton --track origin/develop
    git pull skeleton HEAD
    git status
    ```

1. Review the changes displayed by the `status` command.  Fix any conflicts and
   possibly incorrect auto-merges.

1. After resolving each of the conflicts, `add` your changes to the
   branch, `commit`, and `push` your changes:

    ```console
    git add .github/workflows/build.yml 
    git commit
    git push --force --set-upstream origin lineage/skeleton
    ```

    Note that you may *append* to the default merge commit message
    that git creates for you, but *please do not delete the existing
    content*.  It provides useful information about the merge that is
    being performed.

1. Wait for all the automated tests to pass.

1. Confirm each item in the "Pre-approval checklist" below.

1. Remove any of the checklist items that do not apply.

1. Ensure every remaining checkbox has been checked.

1. Mark this draft pull request "Ready for review".

## ✅ Pre-approval checklist ##

- [x] ✌️ The conflicts in this pull request have been resolved.
- [x] All relevant type-of-change labels have been added.
- [x] All new and existing tests pass.

---

> [!NOTE]
> You are seeing this because one of this repository's maintainers has
> configured [Lineage] to open pull requests.

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

**Review by dv4harr10 — APPROVED** [review]

**Review by dav3r — APPROVED** [review]

**Review by mcdonnnj — APPROVED** [review]
LGTM ✔

**Comment by dav3r** [comment]
@mcdonnnj or @jsf9k Can one of you please add a quick blurb here about why the [lint job](https://github.com/cisagov/skeleton-packer/actions/runs/11605105397/job/32314989369?pr=376) is failing and why that's ok here?

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -40,6 +50,12 @@ jobs:
           - arm64
           - x86_64
     steps:
+      # Note that a duplicate of this step must be added at the top of
+      # each job.
```
Comment (mcdonnnj) [inline .github/workflows/prerelease.yml]:
```suggestion
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -245,6 +232,12 @@ jobs:
           - arm64
           - x86_64
     steps:
+      # Note that a duplicate of this step must be added at the top of
+      # each job.
```
Comment (mcdonnnj) [inline .github/workflows/build.yml]:
```suggestion
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -177,8 +178,17 @@ jobs:
   test:
     needs:
       - diagnostics
+    permissions:
+      # actions/checkout needs this to fetch code
+      contents: read
     runs-on: ubuntu-latest
     steps:
+      # Note that a duplicate of this step must be added at the top of
+      # each job.
```
Comment (mcdonnnj) [inline .github/workflows/build.yml]:
```suggestion
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -47,6 +57,12 @@ jobs:
           - arm64
           - x86_64
     steps:
+      # Note that a duplicate of this step must be added at the top of
+      # each job.
```
Comment (mcdonnnj) [inline .github/workflows/release.yml]:
```suggestion
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: .github/workflows/prerelease.yml (hunk 1 above)
Comment (mcdonnnj) [reply .github/workflows/prerelease.yml]: Resolved in [5a02fb8](https://github.com/cisagov/skeleton-packer/pull/376/commits/5a02fb8244d9bfb8d1a37cd1e2e3deb60bdf03cc).

File: .github/workflows/build.yml (hunk 2 above)
Comment (mcdonnnj) [reply .github/workflows/build.yml]: Resolved in [5a02fb8](https://github.com/cisagov/skeleton-packer/pull/376/commits/5a02fb8244d9bfb8d1a37cd1e2e3deb60bdf03cc).

File: .github/workflows/build.yml (hunk 3 above)
Comment (mcdonnnj) [reply .github/workflows/build.yml]: Resolved in [5a02fb8](https://github.com/cisagov/skeleton-packer/pull/376/commits/5a02fb8244d9bfb8d1a37cd1e2e3deb60bdf03cc).

File: .github/workflows/release.yml (hunk 4 above)
Comment (mcdonnnj) [reply .github/workflows/release.yml]: Resolved in [5a02fb8](https://github.com/cisagov/skeleton-packer/pull/376/commits/5a02fb8244d9bfb8d1a37cd1e2e3deb60bdf03cc).

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -191,35 +199,9 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ steps.setup-env.outputs.python-version }}
-      - uses: actions/cache@v3
-        env:
-          BASE_CACHE_KEY: "${{ github.job }}-${{ runner.os }}-\
-            py${{ steps.setup-python.outputs.python-version }}-\
-            packer${{ steps.setup-env.outputs.packer-version }}-"
+      - uses: hashicorp/setup-packer@v3
```
Comment (mcdonnnj) [inline .github/workflows/build.yml:215]: You've completely removed the `actions/cache` step here in https://github.com/cisagov/skeleton-packer/commit/8666ad511a34280248950a371135b731aa354b83 and I'm guessing that wasn't intentional.

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -257,35 +246,9 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ steps.setup-env.outputs.python-version }}
-      - uses: actions/cache@v3
-        env:
-          BASE_CACHE_KEY: "${{ github.job }}-${{ runner.os }}-\
-            py${{ steps.setup-python.outputs.python-version }}-\
-            packer${{ steps.setup-env.outputs.packer-version }}-\
-            tf-${{ steps.setup-env.outputs.terraform-version }}-"
+      - uses: hashicorp/setup-packer@v3
```
Comment (mcdonnnj) [inline .github/workflows/build.yml:275]: Same thing here.

File: .github/workflows/prerelease.yml [hunk 7]
```diff
@@ -52,35 +66,9 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ steps.setup-env.outputs.python-version }}
-      - uses: actions/cache@v3
-        env:
-          BASE_CACHE_KEY: "${{ github.job }}-${{ runner.os }}-\
-            py${{ steps.setup-python.outputs.python-version }}-\
-            packer${{ steps.setup-env.outputs.packer-version }}-\
-            tf-${{ steps.setup-env.outputs.terraform-version }}-"
+      - uses: hashicorp/setup-packer@v3
```
Comment (mcdonnnj) [inline .github/workflows/prerelease.yml:82]: Same thing here.

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -59,35 +73,9 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ steps.setup-env.outputs.python-version }}
-      - uses: actions/cache@v3
-        env:
-          BASE_CACHE_KEY: "${{ github.job }}-${{ runner.os }}-\
-            py${{ steps.setup-python.outputs.python-version }}-\
-            packer${{ steps.setup-env.outputs.packer-version }}-\
-            tf-${{ steps.setup-env.outputs.terraform-version }}-"
+      - uses: hashicorp/setup-packer@v3
```
Comment (mcdonnnj) [inline .github/workflows/release.yml:89]: Same thing here.

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 5 above)
Comment (jsf9k) [reply .github/workflows/build.yml:215]: Please see commit [5a6801b](https://github.com/cisagov/skeleton-packer/pull/376/commits/5a6801b6de42f7126233ea0ec0582e6208d2078c).

File: .github/workflows/build.yml (hunk 6 above)
Comment (jsf9k) [reply .github/workflows/build.yml:275]: Please see commit [5a6801b](https://github.com/cisagov/skeleton-packer/pull/376/commits/5a6801b6de42f7126233ea0ec0582e6208d2078c).

File: .github/workflows/prerelease.yml (hunk 7 above)
Comment (jsf9k) [reply .github/workflows/prerelease.yml:82]: Please see commit [5a6801b](https://github.com/cisagov/skeleton-packer/pull/376/commits/5a6801b6de42f7126233ea0ec0582e6208d2078c).

File: .github/workflows/release.yml (hunk 8 above)
Comment (jsf9k) [reply .github/workflows/release.yml:89]: Please see commit [5a6801b](https://github.com/cisagov/skeleton-packer/pull/376/commits/5a6801b6de42f7126233ea0ec0582e6208d2078c).

**Comment by jsf9k** [comment]
I think this PR may need a PR similar to cisagov/skeleton-ansible-role#208 to be in place before it can succeed in order to get around the error from the `pip-audit` `pre-commit` hook.  @mcdonnnj, what do you think?

**Comment by jsf9k** [comment]
> I think this PR may need a PR similar to [cisagov/skeleton-ansible-role#208](https://github.com/cisagov/skeleton-ansible-role/pull/208) to be in place before it can succeed in order to get around the error from the `pip-audit` `pre-commit` hook. @mcdonnnj, what do you think?

Please see commits [26a8baf](https://github.com/cisagov/skeleton-packer/pull/376/commits/26a8bafe25f49a099a07342a1539e4dd6eb60095) and [19fbaf3](https://github.com/cisagov/skeleton-packer/pull/376/commits/19fbaf3e1a6da801ce796509b8187ff1aa541d43), as well as cisagov/skeleton-generic#196 and cisagov/skeleton-ansible-role#209.  The commit in this PR gets rid of the `pip-audit` error from the linter, and the two PRs are to keep the other skeleton repos in sync with this one.

**Comment by jsf9k** [comment]
@mcdonnnj - After commit a00c336 `pip-audit` finds a vulnerability in `ansible-core==2.17.6`.  I am confused as to why this does not show up in cisagov/skeleton-ansible-role#209.

[This](https://osv.dev/vulnerability/GHSA-99w6-3xph-cx78) is the vulnerability `pip-audit` finds for `ansible-core` versions 2.16.13, 2.17.6, and even 2.18.0.

File: .github/workflows/prerelease.yml (hunk 7 above)
Comment (jsf9k) [reply .github/workflows/prerelease.yml:82]: Is there a reason why this conversation isn't resolved?  I think I took care of it just as was done in `release.yml`.

Comment (mcdonnnj) [reply .github/workflows/prerelease.yml:82]: Nope, I must've just missed this conversation for some reason when I went through. Sorry about that!

**Review by dav3r — APPROVED** [review]

**Review by dav3r — APPROVED** [review]

**Comment by jsf9k** [comment]
@mcdonnnj - Around Halloween you said that you wanted to rebase #378 on top of this PR when it was ready.  It's ready now.

**Comment by jsf9k** [comment]
> @mcdonnnj - Around Halloween you said that you wanted to rebase #378 on top of this PR when it was ready. It's ready now.

@mcdonnnj - Reminder to rebase when you have time.  I'd like to kick off Lineage for this repo early next week.

**Comment by jsf9k** [comment]
These changes are included in #378.

**Review by mcdonnnj — APPROVED** [review]

**Pull request merged**